lib/uplink/uplink.go

```diff
@@ -74,6 +78,9 @@ func (cfg *Config) setDefaults(ctx context.Context) error {
 	} else if cfg.Volatile.MaxMemory.Int() < 0 {
 		cfg.Volatile.MaxMemory = 0
 	}
+	if cfg.log == nil {
+		cfg.log = zap.L()
```

Comment:
Note, this shouldn't default to logging on Java nor C bindings, since we don't know who their stdout/stderr buffering is setup.

Reply:
so how do we get around the issue of nil pointer exceptions if we do not give it a default value?

Reply:
You default it to `zap.NewNop()`.

Reply:
done